src/metabase/automagic_dashboards/core.clj

```diff
@@ -378,39 +384,46 @@
                  (str/starts-with? name "id_")
                  (str/ends-with? name "_id"))))))
 
+(defn- field-spec-pred
```

Comment:
what does `spec` mean in this matcher and the `named-spec-pred` and why is it absent from `max-cardinality-pred`?

It's also weird that the `field-spec-pred` matches either on the type or the name, and then there's another matcher to match on the regex.

Can these be broken out to `field-type-matcher`, `name-exact-matcher`, `name-regex-matcher`, `max-cardinality-matcher` or something like that?

Reply:
Refactored to consistent naming and added more branch testing.